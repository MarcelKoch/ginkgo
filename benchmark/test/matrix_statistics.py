#!/usr/bin/env python3
import test_framework

# check that all input modes work:
# parameter
test_framework.compare_output(
    ["-input", '[{"size": 100, "stencil": "7pt"}]'],
    expected_stdout="matrix_statistics.simple.stdout",
    expected_stderr="matrix_statistics.simple.stderr",
)

# stdin
test_framework.compare_output(
    [],
    expected_stdout="matrix_statistics.simple.stdout",
    expected_stderr="matrix_statistics.simple.stderr",
    stdin='[{"size": 100, "stencil": "7pt"}]',
)

# input file
test_framework.compare_output(
    ["-input", str(test_framework.sourcepath / "input.mtx.json")],
    expected_stdout="matrix_statistics.simple.stdout",
    expected_stderr="matrix_statistics.simple.stderr",
)

# input matrix file
test_framework.compare_output(
    ["-input_matrix", str(test_framework.matrixpath)],
    expected_stdout="matrix_statistics.matrix.stdout",
    expected_stderr="matrix_statistics.matrix.stderr",
)
