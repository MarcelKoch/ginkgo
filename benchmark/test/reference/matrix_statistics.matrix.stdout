[
    {
        "filename": "",
        "problem": {
            "rows": 36,
            "columns": 36,
            "nonzeros": 208,
            "row_distribution": {
                "min": 4,
                "q1": 4.5,
                "median": 6.0,
                "q3": 7.0,
                "max": 9,
                "mean": 5.777777777777778,
                "variance": 2.061728395061728,
                "skewness": 0.3366362745126052,
                "kurtosis": 2.0507009932231366,
                "hyperskewness": 1.9165991338199193,
                "hyperflatness": 6.0545648993883665
            },
            "col_distribution": {
                "min": 4,
                "q1": 4.5,
                "median": 6.0,
                "q3": 7.0,
                "max": 9,
                "mean": 5.777777777777778,
                "variance": 2.061728395061728,
                "skewness": 0.3366362745126052,
                "kurtosis": 2.0507009932231366,
                "hyperskewness": 1.9165991338199193,
                "hyperflatness": 6.0545648993883665
            }
        },
        "rows": 36,
        "cols": 36,
        "nonzeros": 208
    }
]
